### PR TITLE
[REVIEW BEFORE MERGING] [PartDesign] [Hole] Fixed hole error caused by special characters in windows username

### PR DESCRIPTION
**Pull request by howie-j**
### I am not a c++ developer, please check/fix before merging.. 

The PartDesign::Hole tool fails to read all "Mod/PartDesign/Resources/Hole/xxxxx.json" files if your windows username contains special characters.

[Forum thread: Hole tool fails with special characters in the username](https://forum.freecadweb.org/viewtopic.php?f=19&t=71219)

@wwmayer [fixed a similar issue in techdraw with this commit,](https://github.com/FreeCAD/FreeCAD/commit/ff01905f0b)
so i attempted to fix this issue for the hole tool [with the info from this post.](https://forum.freecadweb.org/viewtopic.php?f=35&t=71220#p619523)

Compiles and works on linux, but not tested on windows.

------------------------------------------------

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [x]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: src/Mod/PartDesign/App/FeatureHole.cpp [hunk 1]
```diff
@@ -2295,7 +2297,8 @@ void Hole::readCutDefinitions()
         for (const auto& f : files) {
             if (f.extension() == "json") {
                 try {
-                    std::ifstream input(f.filePath());
+                    Base::FileInfo fi(f.filePath());
```
Comment (wwmayer) [inline src/Mod/PartDesign/App/FeatureHole.cpp:2300]: `f` already is a FileInfo object. So, no need to create a tmp. object.

Comment (howie-j) [reply src/Mod/PartDesign/App/FeatureHole.cpp]: Thanks, [74f7154](https://github.com/FreeCAD/FreeCAD/pull/7383/commits/74f715431e534134984abd331e4586b981bc56af) is this correct now?

**Comment by wwmayer** [comment]
Merged.

**Comment by luzpaz** [comment]
Congrats @howie-j on your 1st PR and successful merge. As always, thanks again wmayer!

**Pull request closed without merging**